extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpDefaultServicesExtension.java

```diff
@@ -14,27 +14,68 @@
 import org.eclipse.edc.identitytrust.SecureTokenService;
+import org.eclipse.edc.jwt.TokenGenerationServiceImpl;
 import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
 import org.eclipse.edc.runtime.metamodel.annotation.Provider;
+import org.eclipse.edc.runtime.metamodel.annotation.Setting;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.security.KeyPairFactory;
 import org.eclipse.edc.spi.system.ServiceExtension;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
 
+import java.security.KeyPair;
+import java.time.Clock;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
 @Extension("Identity And Trust Extension to register default services")
 public class IatpDefaultServicesExtension implements ServiceExtension {
 
+    @Setting(value = "Alias of private key used for signing tokens, retrieved from private key resolver", defaultValue = "A random EC private key")
+    public static final String STS_PRIVATE_KEY_ALIAS = "edc.iam.sts.privatekey.alias";
+    @Setting(value = "Alias of public key used for verifying the tokens, retrieved from the vault", defaultValue = "A random EC public key")
+    public static final String STS_PUBLIC_KEY_ALIAS = "edc.iam.sts.publickey.alias";
     // not a setting, it's defined in Oauth2ServiceExtension
     private static final String OAUTH_TOKENURL_PROPERTY = "edc.oauth.token.url";
+    @Setting(value = "Self-issued ID Token expiration in minutes. By default is 5 minutes", defaultValue = "" + IatpDefaultServicesExtension.DEFAULT_STS_TOKEN_EXPIRATION)
+    private static final String STS_TOKEN_EXPIRATION = "edc.iam.sts.token.expiration"; // in minutes
+
+    private static final int DEFAULT_STS_TOKEN_EXPIRATION = 5;
+    
+    @Inject
+    private KeyPairFactory keyPairFactory;
+
+    @Inject
+    private Clock clock;
 
     @Provider(isDefault = true)
     public SecureTokenService createDefaultTokenService(ServiceExtensionContext context) {
         context.getMonitor().info("Using the Embedded STS client, as no other implementation was provided.");
+        var keyPair = keyPairFromConfig(context);
+        var tokenExpiration = context.getSetting(STS_TOKEN_EXPIRATION, DEFAULT_STS_TOKEN_EXPIRATION);
+
 
         if (context.getSetting(OAUTH_TOKENURL_PROPERTY, null) != null) {
             context.getMonitor().warning("The property '%s' was configured, but no remote SecureTokenService was found on the classpath. ".formatted(OAUTH_TOKENURL_PROPERTY) +
                     "This could be an indicator of a configuration problem.");
         }
 
-        return new EmbeddedSecureTokenService();
+        return new EmbeddedSecureTokenService(new TokenGenerationServiceImpl(keyPair.getPrivate()), clock, TimeUnit.MINUTES.toSeconds(tokenExpiration));
+    }
+
+    private KeyPair keyPairFromConfig(ServiceExtensionContext context) {
```

Comment:
Maybe the logic of this block should be in `KeyPairFactoryImpl` and thus have the `KeyPairFactory` exposing only a single method, WDYT ?

```java
@ExtensionPoint
@FunctionalInterface
public interface KeyPairFactory {
    Result<KeyPair> generate(ServiceExtensionContext context);
}
```

Reply:
The `KeyPairFactory` should be able to generate a key pair from aliases right? not sure passing the `ServiceExtensionContext` solve this. In this way we can reuse the `KeyPairFactory` here and in other place like `TransferDataPlaneCoreExtension`